woocommerce/woocommerce
# Add tracks to CYS assembler-hub and hide pages sidebar screen

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #40124.

Add tracks to CYS Assembler Hub:

- % of users that click on “Take a tour” vs. “Skip” and close button.
- % of users that complete the onboarding tour.
- % of users that click on each item — Logo, colors, fonts, header, footer, homepage.
- % of users that click on “Editor”.
- - % of users that click on “Styles”.
- % of users that click on “Done”.


Hide the "pages" sidebar screen since we don't need it for MVP.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


1. Use WP 6.3 
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper and enable `customize-store` feature flag
4. Open your browser developer console and run `localStorage.setItem( 'debug', 'wc-admin:*' );` to set debug for tracks
5. Go to `WooCommerce > Home`
6. Click on `Customize your store` task 
7. Click on `Assembler Hub` button
8. Observe that a welcome modal is shown
9. Click on `Skip` button
10. Observe that `wcadmin_customize_your_store_assembler_hub_tour_skip` event is recorded.
11. Delete `woocommerce_customize_store_onboarding_tour_hidden` option (`wp option delete woocommerce_customize_store_onboarding_tour_hidden`)
12. Reload the Assembler Hub page
13. Click on `Take a tour` button
14. Observe that `wcadmin_customize_your_store_assembler_hub_tour_start ` event is recorded.
15. Click on the close button
16. Observe that `wcadmin_customize_your_store_assembler_hub_tour_close` event is recorded.
17. Delete `woocommerce_customize_store_onboarding_tour_hidden` option (`wp option delete woocommerce_customize_store_onboarding_tour_hidden`)
18. Complete the onboarding tour
19.  Observe that `wcadmin_customize_your_store_assembler_hub_tour_complete` event is recorded.
20. Click on `Editor` link
21. Observe that `wcadmin_customize_your_store_assembler_hub_editor_link_click` event is recorded with source = `main`
22. Click on `Change the color palette`
23. Observe that `customize_your_store_assembler_hub_sidebar_item_click` event is recorded with item = `color-palette`
24.  Click on `Editor` link
25.  Observe that `wcadmin_customize_your_store_assembler_hub_editor_link_click` event is recorded with source = `color-palette`
26. Select a color
27. Click on `Save` button
28. Observe that `wcadmin_customize_your_store_assembler_hub_save_click` event is recorded with source = `color-palette`
29. Go back
30. Click on `Done` button
31. Observe that `wcadmin_customize_your_store_assembler_hub_done_click` event is recorded 



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Add tracks to CYS assembler-hub and hide pages sidebar screen 

#### Comment 

</details>
